src/components/header.component.vue

```diff
@@ -0,0 +1,395 @@
+          <div class="SearchHistory__Header">
+            <h3 class="SearchHistory__Text">추천 검색어</h3>
+          </div>
+          <div class="RecommendSearchKeywordList__Container">
+            <button
+              class="CandidateItem__Button"
+              v-for="recommended in DummyData.recommended"
+              :key="recommended"
+            >
+              <div class="CandidateText">{{ recommended }}</div>
+            </button>
+          </div>
+          <div class="Divider__Container">
+            <hr color="#f8f8f8" class="Divider__Line" />
+          </div>
+          <div class="SearchHistory__HeaderRow">
+            <h3 class="SearchHistory__Text">
+              인기 검색어
+              <div class="SearchHistory__GeneratedAt">
+                <img src="../assets/Time_icon.svg" />
+                <div class="SearchHistory__GeneratedAtText">13:15 기준</div>
+              </div>
+            </h3>
+          </div>
+          <div class="PopularSearchKeywordList__Row">
+            <div class="PopularSearchKeywordList__Column">
+              <div
+                class="CandidateItem__Container"
+                v-for="(Item, index) in DummyData.popular.slice(-5)"
+                :key="DummyData.popular[index]"
+              >
+                <span class="CadidateItem__Number" v-if="index < 5">{{
+                  index + 1
+                }}</span>
+                <span class="CandidateItem__Text" v-if="index < 5">{{
+                  Item
+                }}</span>
+              </div>
+            </div>
+            <div class="PopularSearchKeywordList__Column">
+              <div
+                class="CandidateItem__Container"
+                v-for="(Item, index) in DummyData.popular.slice(5)"
+                :key="DummyData.popular[index]"
+              >
+                <span class="CadidateItem__Number">{{ index + 6 }}</span>
+                <span class="CandidateItem__Text">{{ Item }}</span>
+              </div>
+            </div>
+          </div>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'Header',
+  data: () => {
+    return {
+      isShowCategory: false,
+      isShowCartText: true,
+      DummyData: {
+        recommended: [
+          '오늘의 특가',
+          '아이패드0원',
+          'TOP20',
+          '1+1클래스',
+          '여름 취미 추천',
+          '감사제 특가',
+          '10만원할인',
+        ],
+        popular: [
+          '아이패드',
+          '일러스트',
+          '이모티콘',
+          '드로잉',
+          '포토샵',
+          '뜨개질',
+          '레진',
+          '주식',
+          '포토크리에이트',
+          '가전주부 PICK',
+        ],
+      },
+    }
+  },
+  computed: {
+    SearchInputHandler: function() {
+      if (this.isShowCategory == true) {
+        return {
+          width: '676px',
+          margin: 'auto',
+          'align-items': 'center',
+        }
+      } else {
+        return {
+          width: '420px',
+        }
+      }
+    },
```

Comment:
자바스크립트는 대부분, 사실상 모든 프로젝트에서 카멜케이스를 사용합니다.
따라서 소문자로 시작해주시고, 자바스크립트 객체 속성에서 value에 함수가 들어가면 축약이 가능합니다.
```js
// 위, 아래 같습니다.

const obj = {
  a: function() {
   console.log('hi')
  }
}

const obj = {
  a() {
   console.log('hi')
  }
}
```
그리고 이 속성의 역할을 봤을 때 이름이 적절하지 않습니다.
Handler는 보통 이벤트 핸들러 함수에 사용합니다.
이 경우에는 searchInputStyle 이라고 짓는 것이 제일 좋아보입니다.

자바스크립트에서는 == 과 === 이 있습니다. 둘의 동작이 다릅니다. == 을 의도적으로 쓸 경우는 거의 없다고 보시면 됩니다.
그리고 is 가 맨앞에 붙은 변수는 대체로 Boolean 값을 담은 변수로 생각합니다.
따라서 isShowCategory는 true 아니면 false로 생각할 수 있기 때문에, true 와 비교할 필요가 없습니다.

마지막으로 스타일은 클래스를 통해서 변경해주는 것이 바림직한 방향입니다.
스타일과 자바스크립트 코드를 좀 더 분리할 수 있습니다. 이거는 나중에 회상이나 대면으로 알려드릴게요!

```suggestion
    searchInputStyle() {
      if (this.isShowCategory) {
        return {
          width: '676px',
          margin: 'auto',
          'align-items': 'center',
        }
      } else {
        return {
          width: '420px',
        }
      }
    },
```